### PR TITLE
Fix/behaviour

### LINKED ISSUE #9: Modify left and right behaviour
- [x] Left click -> should display the details
- [x] Right click or 3 dots-> open context menu with following options (options should just open a TODO alert except the assign OU should open current dialog):
   - [x] Show Details -> It is implemented
   - [x] Assign OU Data Capture -> It is implemented
   - [x] Assign OU Output -> It is implemented
   - [x] Assign roles -> TODO alert
   - [x] Assign groups -> TODO alert
   - [x] Edit User -> TODO alert

### PR DESCRIPTION
**Pull request by MurhafSousli**
closes #9

### CODE REVIEW

**Comment by adrianq** [comment]
@tokland I am adding you as a reviewer. Can you please have a look at this PR when you are back to work? I will also do a review after yours...

File: src/List/ContextActions.js [hunk 1]
```diff
@@ -9,14 +9,19 @@ config.i18n.strings.add('assignToOrgUnits');
 
 const contextActions = Action.createActionsFromNames([
   'details',
-  'assignToOrgUnits'
+  'assignToOrgUnits',
```
Comment (tokland) [inline src/List/ContextActions.js]: Do you see the icons in the app? I needed to change all "*" to "**" in webpack.config.js proxy routes for the proxy to work (https://github.com/webpack/webpack-dev-server/issues/458#issuecomment-219297105)

**Comment by tokland** [comment]
Btw, did you see the icons? I had to replace `*` by `**` in webpack proxied urls (https://github.com/webpack/webpack-dev-server/issues/458#issuecomment-219297105)

File: src/List/ContextActions.js [hunk 2]
```diff
@@ -8,29 +8,52 @@ config.i18n.strings.add('details');
 config.i18n.strings.add('assignToOrgUnits');
 
 const contextActions = Action.createActionsFromNames([
-    'details',
```
[the remark this answers is not in the document]
Comment (MurhafSousli) [reply src/List/ContextActions.js]: @tokland I didn't intentionally want to change the spaces actually, but I had to hit the organize code shortcut before committing anyway, and It turned out weather I choice 2 or 4 spaces, git was seeing it as total change, so I went with 2 spaces (which I use generally use)

**Comment by MurhafSousli** [comment]
@tokland I added material fonts manually in `index.html` to fix the icon for dev.

File: src/List/ContextActions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrianq) [reply src/List/ContextActions.js]: @tokland what do you use in general? four spaces, right? no tabs, correct? This is a sort of a fork from https://github.com/EyeSeeTea/organisationUnit-User-Assigment. In there, we were using four spaces so probably it is better if we keep the 4 spaces conventions. I definitively would like the three of us to use the same convention

File: src/i18n/i18n_module_en.properties [hunk 3]
```diff
@@ -341,6 +341,9 @@ data_set=Data Set
 intro_data_set=Create, update, view and delete data sets and custom forms. A data set is a collection of data elements for which data is entered.
 
 assignToOrgUnits=Assign to organisation units
+assignRoles=Assign roles
```
[the remark this answers is not in the document]
Comment (adrianq) [reply src/i18n/i18n_module_en.properties]: @MurhafSousli Can you please create a card for this?

[the remark this answers is not in the document]
Comment (MurhafSousli) [reply src/i18n/i18n_module_en.properties]: @adrianq sure

File: src/List/ContextActions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tokland) [reply src/List/ContextActions.js]: Personally, with JS, I like 2-spaces-no-tabs. But for existing projects let's keep whatever we had, here 4-spaces-no-tabs.

**Comment by adrianq** [comment]
@tokland are you fine with this PR? any pending change or something to discuss?

**Comment by tokland** [comment]
I don't see any changes commited from the original review, am I missing something?

> I added material fonts manually in index.html to fix the icon in dev.

Did you try the proposal in my comment? it seemed easier to fix webpack.config.js.

**Comment by MurhafSousli** [comment]
@tokland Sorry I didn't notice that you requested a change. yea it is much better like this!

**Comment by tokland** [comment]
What was the final decision about the spaces? the PR still has the mixture 2 (new) / 4 (old)

**Comment by MurhafSousli** [comment]
@tokland 4 spaces right! I did 4 in the next PR, but I didn't manage to finish it

**Comment by MurhafSousli** [comment]
@adrianq sorry for the delay, I added 4 spaces to all files

**Comment by adrianq** [comment]
Thanks @MurhafSousli What about my second comment?

**Comment by MurhafSousli** [comment]
@adrianq For the context menu, we already have one of them which opens the dialog.

But which property does the current **Assign OU** refer to?
Does it refer to **Assign OU Data Capture**?  or to **Assign OU Output**?

_How am I supposed to know which one?_

I can only think about duplicating this field and rename them to

 - Assign to organization units data capture
 - Assign to organization units output

Assuming that we get the **OU data capture** from `{ fields: ":all,assignToOrgUnits[id,path,displayName]" };`

What should I use for **OU data output**?  
`{ fields: ":all,assignToOrgUnitsOutput[id,path,displayName]" };` didn't work

What is the field name? where can I find this information?

---
After checking your last commit again, I found out that my code was correct except the field was `userGroups` instead of `userGroups[id,displayName,publicAccess]`

This is so frustrating, Am I suppose to keep guessing like this? where is the documentation? The one you gave me is for the REST API. and we are not using it!

**Comment by MurhafSousli** [comment]
@adrianq PR is ready

**Comment by adrianq** [comment]
@MurhafSousli Sorry for being picky but there a couple of things I would like you to polish:
- If you look at the [files comparison](https://github.com/EyeSeeTea/user-app/pull/11/files), there are a lot of files and bits of codes that aren't actually changed, just the format is changing. This may cause some issues (in general it is not considered a good practice): it is more difficult for us to review the code, if in the future we would like to compare it with the original app (organisation user assignment app) it is gonna be more difficult, if in the future I want to see the author of a line because is causing problems or whatever it is gonna be a nightmare, mixing formats, etc. Can you please clean up the code and just commit actual changes?
- There is some code that doesn't have anything to do with this task https://github.com/EyeSeeTea/user-app/issues/9. For instance these lines https://github.com/EyeSeeTea/user-app/pull/11/files#diff-3f783c9074ac4c268b871dac8a540b69R64 I know you will be using this code in the next PR but, in any case, this code should be part of this [PR](https://github.com/EyeSeeTea/user-app/pull/14).

Next tasks are much more difficult and can be a nightmare for the two of us if we are not particularly clean with the PRs. Hopefully, these changes shouldn't take you long. It would be great if we can close these PRs between tomorrow and the day after.

**Comment by MurhafSousli** [comment]
@adrianq I am sorry for the difficulties you are having in reviewing this PR. But do you remember when we talked about the 4 spaces? (A couple of comments above)

 > @MurhafSousli This code looks good to me. A couple of things...
Can you please change the indentation of the new code to 4 spaces? I know that you changed it in the next PR but if you can change it here as well it would definitely help to see the difference. Also, it shouldn't cause you any major conflict in your next PR.

So this is the result of adding 4 spaces. I literally sat the spaces to 4 and then I applied code organizing to all files.

As you see I am not committing all the changes in one giant commit. I made separate commits for this purpose. IMHO I think you should not review the code from the total file changes. instead, it is easier to review each commit on its own. The lines you highlighted are related to the **Add 4 space commit** (https://github.com/EyeSeeTea/user-app/pull/11/commits/4abdafae65b352ab75fb5fe62150ae963ac562a6) I believe.

If the code that was being pushed in earlier PRs was being pushed in a correct format, we wouldn't have seen these annoying changes. but code with extra spaces and wrong formatting will cause such issue and it needs to be fixed sooner or later. otherwise, we are keeping the unfixed code with us from PR to PR.

***

 > There is some code that doesn't have anything to do with this task #9

Sorry, I don't know how this happened, I probably made mistake while checking out between branches and forgot what branch I was working on. this mistake was produced in the commit **Add group to the detail panel**(https://github.com/EyeSeeTea/user-app/pull/11/commits/1371215ac4dab2b8804ebfc5e954bda59036f136)

I honestly don't know how to clean up at this point due to my limited git skills. I don't think I can go back to that state. But I can undo the last commit and push it again into multiple commits if you want.

Would you like remove this PR and start a new one. or if you know a better way, can you guide me through the steps? thanks.

**Pull request merged**